include/coro/queue.hpp

```diff
@@ -147,25 +142,17 @@ class queue
     ~queue() {}
 
     queue(const queue&) = delete;
-    queue(queue&& other)
-    {
-        m_waiters  = std::exchange(other.m_waiters, nullptr);
-        m_mutex    = std::move(other.m_mutex);
-        m_elements = std::move(other.m_elements);
-        m_shutting_down = std::move(other.m_shutting_down);
-        m_stopped = std::move(other.m_stopped);
-    }
+    queue(queue&& other) { *this = std::move(other); }
 
     auto operator=(const queue&) -> queue& = delete;
     auto operator=(queue&& other) -> queue&
     {
         if (std::addressof(other) != this)
         {
-            m_waiters  = std::exchange(other.m_waiters, nullptr);
-            m_mutex    = std::move(other.m_mutex);
-            m_elements = std::move(other.m_elements);
-            m_shutting_down = std::move(other.m_shutting_down);
-            m_stopped = std::move(other.m_stopped);
+            m_waiters       = std::exchange(other.m_waiters, nullptr);
```

Comment:
Unfortunately I think the move constructor and move assignment operator are broken without being able to move the `coro::mutex`, we could put it in a `std::unique_ptr<coro::mutex.` to make it properly move-able at the cost of an extra allocation OR just disable moving and force the user to place it into a `std::unique_ptr` if they need to be able to move it around. I really don't know why this compiled on all platforms previously because it should absolutely not have been allowed and I definitely missed it.

Reply:
Yeah, you are right. Just creating a new mutex in the move constructor will break the queue.
I think the main question to answer is whether it is important that the `coro::queue` type is movable?

Since your other collection type - `coro::ring_buffer` - is not movable too, I would tend to deleting the move semantics for `coro::queue` as well.